tests/php/Control/LoginSessionControllerTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace SilverStripe\SessionManager\Tests\Control;
+
+use HttpRequest;
+use SilverStripe\Control\Director;
+use SilverStripe\Control\Middleware\ConfirmationMiddleware\Url;
+use SilverStripe\Control\Session;
+use SilverStripe\Control\Tests\HttpRequestMockBuilder;
+use SilverStripe\Dev\FunctionalTest;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\ORM\FieldType\DBDatetime;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\Security;
+use SilverStripe\Security\SecurityToken;
+use SilverStripe\SessionManager\Control\LoginSessionMiddleware;
+use SilverStripe\SessionManager\Model\LoginSession;
+
+class LoginSessionControllerTest extends FunctionalTest
+{
+    protected static $fixture_file = '../LoginSessionTest.yml';
+
+    use HttpRequestMockBuilder;
+
+    /** @var Session */
+    private $session;
+
+    /** @var HttpRequest */
+    private $request;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $sessionID = $this->idFromFixture(LoginSession::class, 'x1');
+        $this->session = new Session(['activeLoginSession' => $sessionID]);
+        $this->request =  $this->buildRequestMock('/', [], [], null, $this->session);
+        $this->request->method('getIP')->willReturn('192.168.0.1');
+
+        $memberID = $this->idFromFixture(Member::class, 'owner');
+        $this->logInAs($memberID);
+        SecurityToken::enable();
+        $this->session->set('SecurityID', SecurityToken::inst()->getValue());
+    }
+
+    public function testRemove()
+    {
+        $sessionID = $this->idFromFixture(LoginSession::class, 'x3');
+        $token = SecurityToken::inst()->getValue();
+        $response = Director::test(
+            "admin/loginsession/remove/$sessionID?SecurityID=$token",
+            null,
+            $this->session,
+            'DELETE'
+        );
+
+        $this->assertEquals('200', $response->getStatusCode());
+        $body = json_decode($response->getBody(), true);
+        $this->assertEquals(['success' => true, 'message' => 'Successfully logged out of device.'], $body);
+
+        $this->assertEmpty(
+            LoginSession::get()->byID($sessionID),
+            'x3 LoginSession has been deleted'
+        );
+    }
+
+    public function testBadSecurityID()
+    {
+        $sessionID = $this->idFromFixture(LoginSession::class, 'x3');
+        $token = 'whats-CSRF';
+        $response = Director::test(
+            "admin/loginsession/remove/$sessionID?SecurityID=$token",
+            null,
+            $this->session,
+            'DELETE'
+        );
+
+        $this->assertEquals('200', $response->getStatusCode());
+        $body = json_decode($response->getBody(), true);
+        $this->assertEquals(['success' => false, 'message' => 'Something went wrong.'], $body);
```

Comment:
At the risk flogging a dead horse, these tests would be easier to write and much more expressive if we were returning sensible HTTP response codes.

For example, we could be catching the wrong error here. Maybe our fake CSRF token was accepted and we are getting an error response because of some other problem.